src/lib.rs

```diff
@@ -70,31 +72,53 @@ pub struct Violation {
     /// Description of the error.
     message: String,
     /// The location at which the error was detected.
-    position: ViolationPosition,
+    range: ViolationPosition,
 }
 
 impl Violation {
-    pub fn new<T: AsRef<str>>(message: T, position: ViolationPosition) -> Self {
+    pub fn new<T: AsRef<str>>(message: T, range: ViolationPosition) -> Self {
         Self {
             message: String::from(message.as_ref()),
-            position,
+            range,
         }
     }
 
     pub fn from_node<T: AsRef<str>>(message: T, node: &tree_sitter::Node) -> Self {
-        let position = node.start_position();
         Violation::new(
             message,
-            ViolationPosition::LineCol((position.row + 1, position.column + 1)),
+            ViolationPosition::Range(TextRange::new(
+                TextSize::try_from(node.start_byte()).unwrap(),
+                TextSize::try_from(node.end_byte()).unwrap(),
```

Comment:
These unwraps feel a bit scary. Should this instead return `Result<Self>` and be handled with `?` everywhere?

Reply:
What we can do is check the file length as soon as we open it to ensure it's representable as `u32` (i.e. < 4GB), and then we know all these casts/unwraps are fine.

Reply:
Ah, if that's the only time that this can panic, then it should be fine to leave them in, although maybe with a comment to explain why.